### PR TITLE
update to tokio v1

### PR DESCRIPTION
**Pull request by dr-bonez**

### CODE REVIEW

**Comment by gdesmott** [comment]
Thanks for the port!

I'll need this as well, any chance to have this integrated @vorot93 ?

**Comment by thedodd** [comment]
Looks like CI is failing due to:
```
Error: Unable to process command '##[add-path]/Users/runner/.cargo/bin' successfully.
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
```

Pretty simple fix. Updating that CI code as part of the PR will probably expedite acceptance `:)`.

**Comment by vorot93** [comment]
Hey, sorry!

@dr-bonez do you mind fixing the CI?

File: Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 [package]
 name = "tokio-tar"
-version = "0.2.0"
+version = "0.4.0"
```
Comment (vorot93) [inline Cargo.toml:3]: why not 0.3.0?

**Review by vorot93 — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Any update on this? We've for now vendored the crate into our workspace and made it work there but it would be great to just depend on the version from crates.io.

Anything we can do to push this forward?

**Comment by vorot93** [comment]
@thomaseizinger Thanks for the ping!

The only hurdle are failing tests - once fixed I will merge immediately

**Comment by dr-bonez** [comment]
looks like tokio doesn't support `symlink` on windows anymore

**Comment by vorot93** [comment]
Thanks!

**Comment by thomaseizinger** [comment]
Thanks so much for getting this merged!

Any chance we can also get a release to crates.io? :)

**Comment by vorot93** [comment]
0.3.0 is on crates.io

**Pull request merged**